src/main/java/org/opentripplanner/updater/vehicle_sharing/vehicles_positions/VehicleSharingGraphWriterRunnable.java

```diff
@@ -112,7 +112,7 @@ private void addAppearedRentableVehicles(Graph graph) {
         long properlyLinkedVehicles = Optional.ofNullable(vehiclesForSnapshot.get(true)).orElse(0L);
         LOG.info("Currently there are {} properly linked rentable vehicles from snapshot {} in graph",
                 properlyLinkedVehicles, this.snapshotLabel);
-        if (properlyLinkedVehicles > 0) {
+        if (properlyLinkedVehicles > 0 && !this.snapshotLabel.isEmpty()) {
```

Comment:
Plain `HashSet` is not thread-safe - I must've overlooked it earlier. If two runnables will try to add snapshot labels at the same time, it might cause some failure. Maybe it is better to use `HashMap<SharedVehiclesSnapshotLabel, Boolean> isLinked`, which would be initialized to `false` for each label in `graph updater` and then set to `true` in each `runnable`. We would have possibly multiple reads (from health checks), but writes would be made by only one thread per boolean.